flake.nix

```diff
@@ -100,7 +101,7 @@
               formatModule
             ]
             ++ modules;
-        };
+        } // nixosConfig;
```

Comment:
I don't think this is needed. You can use the modules parameter to pass in your customizations. Also would needs to use `imports` so it merges correctly.

Reply:
Sorry, meant to explain further in PR. I've tried passing through my `nixosConfiguration` as a module, and I get the following error:

```shell
nix build -L .\#packages.MTX-SRV
error:
       … while evaluating the attribute 'config.system.build."${(image).config.formatAttr}"'

         at /nix/store/5z81la6av4j0ckp0w1949lxiwjyyykks-source/lib/modules.nix:322:9:

          321|         options = checked options;
          322|         config = checked (removeAttrs config [ "_module" ]);
             |         ^
          323|         _module = checked (config._module);

       … while calling the 'seq' builtin

         at /nix/store/5z81la6av4j0ckp0w1949lxiwjyyykks-source/lib/modules.nix:322:18:

          321|         options = checked options;
          322|         config = checked (removeAttrs config [ "_module" ]);
             |                  ^
          323|         _module = checked (config._module);

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: Could not load a value as a module, because it is of type "configuration", in file /nix/store/5z81la6av4j0ckp0w1949lxiwjyyykks-source/flake.nix. If you do intend to import this configuration, please only import the modules that make up the configuration. You may have to create a `let` binding, file or attribute to give yourself access to the relevant modules.
       While loading a configuration into the module system is a very sensible idea, it can not be done cleanly in practice.
```

I thought my PR might be a good starting point to get this working.

The source is [here](https://github.com/shymega/nixfigs/blob/experimental/nixos-generators/nix/generators.nix)

Reply:
If you just want to generate for your nixos configuration than import nixos-generators.nixosModules.all-formats in your nixos configuration and than reference
`nixosConfigurations.my-machine.config.formats.proxmox` instead. One cannot import a nixos systems like that as a module.

Reply:
Right, okay. Should I close the PR? It would be nice to be able to pass a `nixosConfiguration` to `nixosGenerate`, but I understand (as a maintainer) if you'd rather keep the current functionality.